node/src/accountant/db_big_integer/big_int_db_processor.rs

```diff
@@ -1,43 +1,57 @@
 // Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
 use crate::accountant::checked_conversion;
-use crate::accountant::db_access_objects::payable_dao::PayableDaoError;
 use crate::accountant::db_access_objects::receivable_dao::ReceivableDaoError;
 use crate::accountant::db_big_integer::big_int_divider::BigIntDivider;
-use crate::database::connection_wrapper::ConnectionWrapper;
+use crate::accountant::PayableDaoError;
+use crate::database::rusqlite_wrappers::{ConnectionWrapper, TransactionWrapper};
 use crate::sub_lib::wallet::Wallet;
 use itertools::Either;
-use rusqlite::{Error, Row, Statement, ToSql, Transaction};
+use rusqlite::{Error, Row, Statement, ToSql};
 use std::fmt::{Debug, Display, Formatter};
 use std::iter::once;
 use std::marker::PhantomData;
 use std::ops::Neg;
 
+pub trait BigIntDatabaseProcessor<T>: Debug + Send
+where
+    T: TableNameDAO,
+{
+    fn execute<'a>(
```

Comment:
If I were writing this I might've tried to prevent using traits here. I do feel when we use traits, they end up biting us later. Let me know what you think of using a trait here.

Reply:
I did it in order to acquire power in the position of testing. It allows mocking.